### ISSUE TITLE
--exclude-limit seems like an anti-feature

### ISSUE THREAD

**Issue by ClayShentrup** [rubocop-hq/rubocop#3339]
To be clear, this is not a bug.

Last week I set up Rubocop with a todo file with my new employer. I wondered why I wasn't getting more requests for help sorting out violations, and then I discovered last night to my great surprise that many of the rules were completely disabled in the todo file because they had more than 15 occurrences.

This behavior seems contrary to the entire purpose of --auto-gen-config, which is to put a stake in the ground and ratchet down the errors over time. If I wanted to globally disable a rule, it wouldn't make any sense to put it in a todo file, as this would allow the violation to propagate into new files.

To summarize, I think this behavior violates the "principle of least surprise". It also took me some time to figure out why this was happening and how to get the expected behavior. I can't imagine I'm in the minority here.

Thanks
## RuboCop version

0.41.1

**Comment by metcalf** [comment]
Agreed that the default is surprising. We use this feature but at a value of 1000 by default. The current default is probably sane for small projects but definitely doesn't make sense for a million line codebase.

**Comment by ClayShentrup** [comment]
Thanks @metcalf. What I'm saying is that I can't even see the value of this behavior in a _small_ codebase. Because, if you wanted to disable a feature completely, you could consciously do it in `.rubocop.yml`. The whole _point_ of the todo config is to put a stake in the ground and prevent any more violations. I literally cannot see any reason for `--exclude-limit` to exist. Or at the very least, I would expect it to be unlimited by default, requiring it to be explicitly passed.

**Comment by metcalf** [comment]
Ah, yep. I agree that the default should be unlimited (or at least very
high if unlimited can wreck unexpected havoc for some reason).

**Comment by carmi** [comment]
Is there a way to define a large or larger `--exclude-limit` in a config file rather than having to pass it as an argument with each invocation? This would somewhat help to keep the exclude limit higher if not unlimited.

**Comment by bbatsov** [comment]
I'll defer to @jonas054 on this one, as I don't recall what was the reasoning behind the current implementation. I'm open to concrete suggestions about how to improve this.

**Comment by jonas054** [comment]
@ClayShentrup Makes a good point. It does seem like an anti-feature. I've tried to dig through the GitHub history to find what the motivation was originally, but I didn't find anything. My guess is that we wanted a short and readable `.rubocop_todo.yml`, a goal that I now see as less important than being able to improve the inspected code incrementally.

If somebody wants to change the default to unlimited, that's fine with me.

**Comment by andyw8** [comment]
This was recently mentioned in https://medium.com/@scottm/rubocop-in-legacy-projects-part-1-todos-and-todonts-877ace9f23b7 by @scottmatthewman.
